### ISSUE TITLE
Encountering problems with MATLAB code

### ISSUE THREAD

**Issue by twilight-kk** [sevagh/Real-Time-HPSS#2]
I used the code and signals provided by the blogger to run HPSS. m, and the h and p components I obtained were abnormal, especially the first few values, which were very large. What is the reason for the abnormality?

**Comment by sevagh** [comment]
I would guess dividing by small numbers when there's not enough data, or a lot of silence. What type of audio/signal are you using?

**Comment by twilight-kk** [comment]
Yes, you are right. I later discovered that the input signal was incorrect. I replaced it with a new mixed signal and also used my own .wav audio signal, and it was possible to separate it normally.

However, when I plotted the spectra of the separated harmonic (h) and percussive (p) components, I found that their spectra were abnormal. Therefore, I modified the HPSS.m script where win = sqrt(hann(winLen, "periodic")) was used, changing it to win = hann(winLen, "periodic") and also tried using Hamming windows. This adjustment made the spectra look normal, but the amplitude of the h and p in the time domain decreased significantly, meaning their intensity was reduced. Is this a problem related to the total energy change? I am still a bit confused about this issue and would appreciate your advice on how to resolve it. Looking forward to your response!

**Comment by sevagh** [comment]
What does an abnormal spectrum look like? You can post the spectrum here, or the audio file if possible.

**Comment by twilight-kk** [comment]
Thank you for your reply. I am Chinese and currently celebrating the traditional Mid-Autumn Festival, so I have not been able to go to the lab these days and cannot send you the anomaly spectrum in a timely manner. I apologize for the inconvenience. I will send it to you after September 18th, which is Wednesday. Thank you!
